samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -101,10 +102,12 @@ def _can_process_application(self, app):
 
         :param dict app: the application and its properties
         """
+        location = app.properties.get(self.LOCATION_KEY)
         return (self.LOCATION_KEY in app.properties
-                and isinstance(app.properties[self.LOCATION_KEY], dict)
-                and self.APPLICATION_ID_KEY in app.properties[self.LOCATION_KEY]
-                and self.SEMANTIC_VERSION_KEY in app.properties[self.LOCATION_KEY])
+                and isinstance(location, dict)
+                and list(location.keys())[0] not in resolver.DEFAULT_SUPPORTED_INTRINSICS
```

Comment:
why it only checks for the first key?

Reply:
If it's an S3 location, it will only have one dictionary key at this level:
```
Location:
  Ref: parameter
# or
Location:
  Sub: s3://{something}
```
If it has something else, then it's probably not valid.

This brings up a good point though: this doesn't check for all possible intrinsics. I need to fix that.